src/app/products/products.service.ts

```diff
@@ -90,6 +90,9 @@ export class ProductsService {
     return this.http.put(`/products/share/${shareProductId}`, shareProduct);
   }
 
+  getDividends(shareProductId: string): Observable<any> {
+    return this.http.get(`/shareproducts/${shareProductId}/dividends`);
```

Comment:
```suggestion
    return this.http.get(`/shareproduct/${shareProductId}/dividend`);
```
Please check the endpoint, navigation to component gives 404 in API response